sdk/eventgrid/eventgrid/review/eventgrid.api.md

```diff
@@ -99,47 +131,121 @@ export type ContainerRegistryImageDeletedEventData = ContainerRegistryEventData
+    body?: any;
+    properties?: {
+        [propertyName: string]: string;
+    };
+    systemProperties?: {
+        [propertyName: string]: string;
+    };
 }
 
 // @public
-export type EventGridClientOptions = PipelineOptions;
+export interface DeviceTwinInfo {
+    authenticationType?: string;
+    cloudToDeviceMessageCount?: number;
+    connectionState?: string;
+    deviceId?: string;
+    etag?: string;
+    lastActivityTime?: string;
+    properties?: DeviceTwinInfoProperties;
+    status?: string;
+    statusUpdateTime?: string;
+    version?: number;
+    x509Thumbprint?: DeviceTwinInfoX509Thumbprint;
+}
+
+// @public
+export interface DeviceTwinInfoProperties {
+    desired?: DeviceTwinProperties;
+    reported?: DeviceTwinProperties;
+}
+
+// @public
+export interface DeviceTwinInfoX509Thumbprint {
+    primaryThumbprint?: string;
+    secondaryThumbprint?: string;
+}
+
+// @public
+export interface DeviceTwinMetadata {
+    lastUpdated?: string;
+}
+
+// @public
+export interface DeviceTwinProperties {
+    metadata?: DeviceTwinMetadata;
+    version?: number;
+}
 
 // @public
 export class EventGridConsumer {
-    constructor(decoders?: Record<string, CustomEventDataDecoder>);
-    // Warning: (ae-forgotten-export) The symbol "CustomEventDataDecoder" needs to be exported by the entry point index.d.ts
-    //
+    constructor(options?: EventGridConsumerOptions);
     // (undocumented)
-    readonly customDecoders: Record<string, CustomEventDataDecoder>;
-    decodeCloudEvents(encodedEvents: string): Promise<CloudEvent<unknown>[]>;
-    decodeCloudEvents(encodedEvents: object): Promise<CloudEvent<unknown>[]>;
-    decodeEventGridEvents(encodedEvents: string): Promise<EventGridEvent<unknown>[]>;
-    decodeEventGridEvents(encodedEvents: object): Promise<EventGridEvent<unknown>[]>;
+    readonly customDeserializers: Record<string, CustomEventDataDeserializer>;
+    deserializeCloudEvents(encodedEvents: string): Promise<CloudEvent<unknown>[]>;
+    deserializeCloudEvents(encodedEvents: object): Promise<CloudEvent<unknown>[]>;
+    deserializeEventGridEvents(encodedEvents: string): Promise<EventGridEvent<unknown>[]>;
+    deserializeEventGridEvents(encodedEvents: object): Promise<EventGridEvent<unknown>[]>;
+}
+
+// @public
+export interface EventGridConsumerOptions {
+    customDeserializers: Record<string, CustomEventDataDeserializer>;
 }
 
 // @public
 export interface EventGridEvent<T> {
     data: T;
     dataVersion: string;
-    eventTime: Date;
+    eventTime?: Date;
     eventType: string;
-    id: string;
+    id?: string;
     subject: string;
     topic?: string;
 }
 
+// @public
+export class EventGridPublisherClient {
+    constructor(endpointUrl: string, credential: KeyCredential | SignatureCredential, options?: EventGridPublisherClientOptions);
+    readonly apiVersion: string;
+    readonly endpointUrl: string;
+    sendCloudEvents(events: CloudEvent<any>[], options?: SendCloudEventsOptions): Promise<RestResponse>;
```

Comment:
`RestResponse` is kind of ugly and low-level. If there is no specific return type, would it be better to pluck off `_response` and just return that as `HttpResponse`? What is this return value used for exactly? Are customers going to want to be looking at headers/http status?

Reply:
> Are customers going to want to be looking at headers/http status?

It seems unlikely customers would want to do this in general.  Do you think it would be worthwhile to define some interfaces like this?

```
interface SendEventsResponse {
  _response: HttpResponse
}
```

And go from there?  That at least gives us the ability to evolve the return type in a compatible way if we decide there's more information we want to return at a later date?

Reply:
I went ahead and make this change.  Let me know what you think.

Reply:
I like this a lot!